src/field_theory/is_alg_closed/basic.lean

```diff
@@ -115,10 +115,14 @@ theorem of_exists_root (H : ∀ p : polynomial k, p.monic → irreducible p →
  let ⟨x, hx⟩ := H (q * C (leading_coeff q)⁻¹) (monic_mul_leading_coeff_inv hq.ne_zero) this in
  degree_mul_leading_coeff_inv q hq.ne_zero ▸ degree_eq_one_of_irreducible_of_root this hx⟩
 
-lemma degree_eq_one_of_irreducible [is_alg_closed k] {p : polynomial k} (h_nz : p ≠ 0)
+lemma degree_eq_one_of_irreducible [is_alg_closed k] {p : polynomial k}
   (hp : irreducible p) :
   p.degree = 1 :=
-degree_eq_one_of_irreducible_of_splits h_nz hp (is_alg_closed.splits_codomain _)
+begin
+  by_cases h_nz : p = 0,
+  { exfalso, simp [*] at *, },
```

Comment:
this one should probably be fixed by going further up:

```lemma degree_eq_one_of_irreducible_of_splits {p : polynomial L}
 (hp : irreducible p) (hp_splits : splits (ring_hom.id L) p) :
  p.degree = 1 :=
begin
  rcases hp_splits,
  { have := hp.ne_zero, contradiction },
  { apply hp_splits hp, simp }
end
```

I've got to say, this isn't the first time I've seen "non-zero irreducible polynomial" in mathlib :D